crates/abq_utils/src/net_protocol/runners.rs

```diff
@@ -289,38 +289,59 @@ impl From<RawManifest> for Manifest {
 impl Manifest {
     /// Flattens a manifest into [TestSpec]s, preserving the manifest order.
     pub fn flatten(self) -> (Vec<TestSpec>, MetadataMap) {
-        use v0_2::{Group, Test, TestCase, TestOrGroup};
-
+        use v0_2::{Group, TestOrGroup};
         let Manifest { members, init_meta } = self;
         let mut collected = Vec::with_capacity(members.len());
-        let mut queue: VecDeque<_> = members.into_iter().collect();
-        while let Some(test_or_group) = queue.pop_front() {
-            match test_or_group {
-                TestOrGroup::Test(Test { id, meta, .. }) => {
-                    let spec = TestSpec {
-                        // Generate a fresh ID for ABQ-internal usage.
-                        work_id: WorkId::new(),
-                        test_case: self::TestCase(
-                            TestCase {
-                                id,
-                                meta,
-                                focus: None,
-                            }
-                            .into(),
-                        ),
-                    };
-                    collected.push(spec);
+
+        // 🙋 Ayaz told me to reuse this Deque but I don't understand the implications of re-creating where we extend it below
+        let mut queue: VecDeque<v0_2::TestOrGroup> = VecDeque::new();
+        for top_level_test_or_group in members {
+            let group_id = GroupId::new();
+            match top_level_test_or_group {
+                TestOrGroup::Test(test) => {
+                    Self::add_test_to_collected(test, &mut collected, group_id)
                 }
                 TestOrGroup::Group(Group { members, .. }) => {
-                    for member in members.into_iter().rev() {
-                        queue.push_front(member);
+                    queue.clear();
+                    queue.extend(members);
+                    while let Some(test_or_group) = queue.pop_front() {
+                        match test_or_group {
+                            TestOrGroup::Test(test) => {
+                                Self::add_test_to_collected(test, &mut collected, group_id)
+                            }
+                            TestOrGroup::Group(Group { members, .. }) => {
+                                for member in members.into_iter().rev() {
+                                    queue.push_front(member);
+                                }
+                            }
+                        }
                     }
                 }
             }
         }
         (collected, init_meta)
     }
 
+    fn add_test_to_collected(
```

Comment:
a bit of a nit, but because this is one of the few CPU-based things the queue does, want to avoid needless calls

```suggestion
    #[inline]
    fn add_test_to_collected(
```